packages/mllp-server/src/utils.ts

```diff
@@ -46,3 +55,42 @@ export function withErrorHandling<T>(
     });
   };
 }
+
+export interface ParsedHl7Data {
+  message: Hl7Message;
+  cxId: string;
+  patientId: string;
+}
+
+export async function parseHl7Message(rawMessage: Hl7Message): Promise<ParsedHl7Data> {
```

Comment:
This seem pretty central to the MLLP server, so maybe have it on it's own file, like `parser.ts`?